cmd/create/cluster/cmd.go

```diff
@@ -1013,8 +1036,16 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 		// Validate the role names are available on AWS
 		for _, role := range operatorIAMRoleList {
-			name := strings.SplitN(role.RoleARN, "/", 2)[1]
-			err := awsClient.ValidateRoleNameAvailable(name)
+			str, err := arn.Parse(role.RoleARN)
+			if err != nil {
+				r.Reporter.Errorf("Error validating role arn: %v", err)
+				os.Exit(1)
+			}
+			resource := str.Resource
+			m := strings.LastIndex(resource, "/")
```

Comment:
I would recommend extracting this logic to a common util function (like https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/pkg/aws/stshelper.go#L221) in one of the `helper` files.

Ideally, that function would be in https://gitlab.cee.redhat.com/service/ocm-common/, so both CS and ROSA CLI use the very same one, but I assume that can be added to `ocm-common` later.

Reply:
do we have access to https://gitlab.cee.redhat.com/service/ocm-common/?

Reply:
I've invited you to the repo.

Reply:
What I meant was: does the github repo have access to redhat's gitlab? cause I can't access it without VPN.

Reply:
Ok, I understand, now that you mention it I'm not sure we import `ocm-common` in this project.